1MI0800224/week2/week2_task3.cpp

```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+#include <fstream>
+struct pair{
+    int x;
+    int y;
+};
+
+pair createPair(int x, int y){
+    pair newPair={x, y};
+    return newPair;
+}
+
+void InitPair(pair& toInit, int x, int y){
+    toInit={x, y};
+}
+
+pair ReadPairFromFile(std::ifstream& inFile){
+    if(!inFile.is_open()){
+        return {INT_MIN, INT_MAX};
+    }
+    int x, y;
+    inFile>>x>>y;
+    return {x, y};
+    inFile.close();

```

Comment:
Mmm, having a statement after the return will not work unfortunately (see: [here](https://www.stroustrup.com/bs_faq2.html#finally)). The typical concept for cases like this is as follows - whoever opens the file needs to check that the operation succeeded and to also close it when it's no longer needed (same applies for dynamic memory too - whoever allocates it is supposed to free it, otherwise it becomes `fun & games`)